go-controller/pkg/cni/helper_linux.go

```diff
@@ -50,39 +50,68 @@ func moveIfToNetns(ifname string, netns ns.NetNS) error {
 	return nil
 }
 
-func setupNetwork(link netlink.Link, macAddress, ipAddress, gatewayIP string) error {
-	hwAddr, err := net.ParseMAC(macAddress)
+func setupNetwork(link netlink.Link, ifInfo *PodInterfaceInfo) error {
+	hwAddr, err := net.ParseMAC(ifInfo.MAC)
 	if err != nil {
 		return fmt.Errorf("failed to parse mac address for %s: %v", link.Attrs().Name, err)
 	}
 	err = netlink.LinkSetHardwareAddr(link, hwAddr)
 	if err != nil {
-		return fmt.Errorf("failed to add mac address %s to %s: %v", macAddress, link.Attrs().Name, err)
+		return fmt.Errorf("failed to add mac address %s to %s: %v", ifInfo.MAC, link.Attrs().Name, err)
 	}
-	addr, err := netlink.ParseAddr(ipAddress)
+	addr, err := netlink.ParseAddr(ifInfo.IP)
 	if err != nil {
 		return err
 	}
 	err = netlink.AddrAdd(link, addr)
 	if err != nil {
-		return fmt.Errorf("failed to add IP addr %s to %s: %v", ipAddress, link.Attrs().Name, err)
+		return fmt.Errorf("failed to add IP addr %s to %s: %v", ifInfo.IP, link.Attrs().Name, err)
 	}
 
-	gw := net.ParseIP(gatewayIP)
-	if gw == nil {
-		return fmt.Errorf("parse ip of gateway failed")
+	var foundDefault bool
+	for _, route := range ifInfo.Routes {
+		ipaddr, ipnet, err := net.ParseCIDR(route.Dest)
+		if err != nil {
+			return fmt.Errorf("failed to parse pod route %q: %v", route.Dest, err)
+		}
+		ipnet.IP = ipaddr
+
```

Comment:
@dcbw .. just a NIT.. Should L73 to 77, just be

```
	_, ipnet, err := net.ParseCIDR(route.Dest)
		if err != nil {
			return fmt.Errorf("failed to parse pod route %q: %v", route.Dest, err)
		}
```
The `destination` in routes is going to be a network CIDR and ipnet.IP already is a network number and is in the right format for IPRoute()

Reply:
@girishmg updated, PTAL.